openrr-client/src/clients/local_move.rs

```diff
@@ -0,0 +1,152 @@
+use arci::{BaseVelocity, MoveBase, Navigation};
+use k::nalgebra as na;
+use na::Isometry2;
+use serde::{Deserialize, Serialize};
+
+use crate::Error;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct LocalMoveConfig {
+    pub reach_distance_threshold: f64,
+    pub reach_angle_threshold: f64,
+    pub control_frequency: f64,
+    pub linear_gain: f64,
+    pub angular_gain: f64,
+    pub max_linear_vel: f64,
+    pub max_angular_vel: f64,
+}
+
+impl LocalMoveConfig {
+    pub fn try_new<P: AsRef<std::path::Path>>(path: P) -> Result<Self, Error> {
+        let config: LocalMoveConfig = toml::from_str(
+            &std::fs::read_to_string(&path)
+                .map_err(|e| Error::NoFile(path.as_ref().to_owned(), e))?,
+        )
+        .map_err(|e| Error::TomlParseFailure(path.as_ref().to_owned(), e))?;
+        Ok(config)
+    }
+}
+
+pub struct LocalMove {
+    pub nav_client: Box<dyn Navigation>,
+    pub vel_client: Box<dyn MoveBase>,
```

Comment:
Could you try these generics instead of Box<dyn XXX>?

In many cases Box<dyn XXX> implements XXX automatically by #[auto_Impl].

Reply:
Ok, I try it.